### ISSUE TITLE
Grouped Select widget orders groups alphabetically

### ISSUE THREAD

**Issue by philippjfr** [holoviz/panel#3786]
In this example despite Europe being listed before Africa, Africa is listed first:

```python
select = pn.widgets.Select(name='Select', groups={'Europe': ['Greece', 'France'], 'Africa': ['Algeria', 'Congo']})

select
```

<img width="1150" alt="Screen Shot 2022-08-26 at 17 43 29" src="https://user-images.githubusercontent.com/1550771/186942909-c1f823b8-96a5-4db2-8f94-136ff071aeb8.png">

**Comment by mdering** [comment]
I looked into this, and its because bokeh sorts all keys when it dumps the json of the object https://github.com/bokeh/bokeh/blob/a1931d48512a1cb369746025d48344a6ad1834b2/bokeh/core/json_encoder.py#L171
Thus when it encodes the `options` attribute of the selector, they get sorted. I didnt see any obvious solution to this problem, especially since its upstream in bokeh

**Comment by jbednar** [comment]
@mattpap?

**Comment by mattpap** [comment]
It's fixed in bokeh 3.0-rc. bokeh now preserves ordering of data structures. In this case it's insertion order for dicts.
